### PR TITLE
Add a couple of non-canonical ctors

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: src/range.rs [hunk 1]
```diff
@@ -54,6 +54,19 @@ pub fn TextRange(start: TextSize, end: TextSize) -> TextRange {
...
+    /// Creates a zero-length range at the specified offset.
+    pub const fn empty(self, offset: TextSize) -> TextRange {
+        TextRange {
+            start: offset,
+            end: offset,
+        }
+    }
+
+    /// Creates a `TextRange` with specified starting offset and length.
+    pub fn offset_len(start: TextSize, length: TextSize) -> TextRange {
```
Comment (matklad) [inline src/range.rs]: I am not terribly happy about the naming here, but I do thing that we need a ctor with this signature. Doing `TextRange(start, start + len)` on the callsite doesn't work if `start` is a non-trivial expression.

**Comment by matklad** [comment]
I do feel that `TextRange::empty` is much better than `TextRange::at`. To me, the latter makes sense only with combination with `with_len` (and that calls for a dedicated `at_with_len` or somthing function). 

> I agree we need an "offset_len", but I'm not willing to accept that name without exploring the design space a bit more.

I now think that, with `TextRange::empty`, I wouldn't miss `offset_len` that much. Like, we still need it, but we can make progress without.

**Comment by matklad** [comment]
Hm, I am starting to think that maybe `TextRage::offset_len(start, len)` should be spelled as `TextRange::xxx(len) + start`, but I don't know a good name for `xxx`. Like, it should be the dual of `empty`, where start, and not length, is zero. `TextRange::to(offset)`?

**Comment by kjeremy** [comment]
`with_offset`?

**Comment by matklad** [comment]
Are you talking about my `xxx`? That should be exactly the opposite, `with_len`

**Comment by CAD97** [comment]
On a previous branch, I had it as `TextRange::of(text: impl TextSized) => TextRange(0..text.len())`. I don't know how obvious that naming is anymore, but just to throw it back out there.

**Comment by matklad** [comment]
I am more concerned not with the naming, but with the fact that the argument should be a `TextSize` and not `impl TextSized`.

**Pull request merged**